### DISCUSSION TITLE
Stats per major version and per minor version

### DISCUSSION THREAD

**Discussion by Pr0methean** [rust-lang/crates.io#8636] [Feature Requests]
I was recently given ownership of the `zip` crate (per https://github.com/zip-rs/zip-old/issues/446#issuecomment-2067754894) to publish what had previously been my fork `zip_next`, and started publishing under the `zip` name on April 20. I'm interested to know how many downloaders are using versions I've published, because I may be looking for a new job soon and the bragging rights would be highly relevant. But due to the automation necessary to resolve https://github.com/zip-rs/zip2/issues/31 and a backlog of 49 PRs at the time of the takeover, I've ended up releasing 9 versions so far, and crates.io's stats and API only show version-specific data for the latest 5 versions of a crate. And I'd think distinguishing pre-takeover and post-takeover versions would be easy, since the former are 0.x and the latter are 1.x (and there's likely to be a 2.0.0 version in Q2 or Q3 2024 that adds an async API).

Could we please have a way to get stats for the 5 newest major or minor versions separately aggregated, rather than only the last 5 patch versions individually? It'd be the next best thing to having every non-yanked patch version's daily stats tracked and made available perpetually. (I can understand excluding yanked versions, since black hats might use them to measure an attack surface.)

**Comment by syphar** [comment]
Until you have it on crates.io, you always can just fetch the data via API, and aggregate yourself. Alternatively, [fetch the database dump](https://crates.io/data-access).

**Comment by Pr0methean** [comment]
Finally found the API endpoint for that, but `crates_io_api` doesn't have a method to use it. Testing directly with a reqwest client using `version.links.version_downloads`...

**Reply by Turbo87** [reply]
I would strongly suggest using the database dump instead... you can even use https://github.com/dtolnay/db-dump to make it easier to work with

**Reply by syphar** [reply]
It seems to be only about a single crate, so the API feels better sized 😄

**Reply by Pr0methean** [reply]
The dump worked, and it took less time than the API since I didn't have to throttle myself to 1 RPS. It has 90 days of data for all versions.
